### PR TITLE
Enable tree shaking

### PR DESCRIPTION
**Pull request by nickwaelkens**
As a series of goodbye gifts, I want to reduce the front-end bundle size significantly. One of the things that has been bothering us since the beginning is the fact that the entire `ui` library is always loaded, even when we only need a simple `Button` or `Label`. 

Part one of this task is making all "subpackages" tree shakeable, because yes-every icon, visual, ... is also always loaded.

## Manual check
Link this package and check it out in a dummy file that just includes stuff from this package (`@teamleader/ui` still bundles everything, so you can't just include it in your existing projects _yet_)

Eg.

```js
import { 
  IconAddBadgedMediumFilled, 
  IconArrowRightMediumOutline, 
  IconArrowLeftMediumOutline 
} from '@teamleader/ui-icons'; // make sure this is your locally linked version

export default () => IconAddBadgedMediumFilled && IconArrowRightMediumOutline
```

This code yields the following bundles:

Before:
```
foo.ed43e5c7d3d931fdd750.js    263 KiB       0  [emitted]  [big]  foo
```

After:
```
foo.9849665ef408f1fa1d1a.js     11 KiB       0  [emitted]  foo
```

No need for React etc, if you check your compiled file you will see that there are only 2 icons defined, even when importing 3 👍 

To check potential artifacts, `npm publish --dry-run`.

### CODE REVIEW

**Comment by nickwaelkens** [comment]
@ArnaudWeyts `react` and `prop-types` are `peerDependencies`, so if you want it to run locally you'll need to install them both.

```sh
npm i react prop-types --no-save
```

☝️  That way you won't update your `package-lock.json`

**Comment by duivvv** [comment]
Can't we define them as dependencies? It's not a bundle that is created, I don't think it will have impact.

File: package.json [hunk 1]
```diff
@@ -1,51 +1,45 @@
...
     "lodash.upperfirst": "^4.3.1",
-    "prettier": "^1.8.2"
+    "prettier": "^1.16.4"
   },
   "engines": {
     "node": "^8.0.0",
     "npm": "^5.0.0"
   },
   "homepage": "https://github.com/teamleadercrm/ui-icons#readme",
+  "jsnext:main": "es/index.js",
```
Comment (duivvv) [inline package.json]: I did some extra reading and writing on the topic and I think we can safely remove this one and go for module alone.

**Comment by lowiebenoot** [comment]
I would just leave them in the peerDependencies, that seems to be the correct place to me. Any particular reason why you would move them over? @duivvv

**Comment by nickwaelkens** [comment]
I, too, see every other library out there doing it like this - will leave it as `peerDependencies`!

**Comment by duivvv** [comment]
nevermind :) You're right

**Pull request merged**